### DISCUSSION TITLE
Table-like form

### DISCUSSION THREAD

**Discussion by ronaldbalza** [Budibase/budibase#3221] [Help]
Hi everyone, is it possible to have a table-like form? I wanted to make it a row form (like all labels are on top, input field are on the bottom of the label) and then add the data on the table before saving it in a database. Thanks for any help :) Cheers!
![image](https://user-images.githubusercontent.com/37702014/139791772-cf3118f7-2ebf-4f6d-b4d1-c026a29d610f.png)

**Comment by mjashanks** [comment]
Hi There. 

Yes, this is possible. I've built it and attached my app export so you can see exactly how it's done.

Key Points
1. Component Structure - see screenshot below
2. The Container's `Direction` should be `Row`, and `Horiz. Align` should be `Left`
3. Your form must be inside the data provider so that the provider knows to refresh after you have saved the row.

**Component Structure**
![image](https://user-images.githubusercontent.com/3524181/139839180-2393c540-6572-4d45-b3ba-3cf19183b561.png)

**App Export**
[Table Like Form-export-1635852719881.txt](https://github.com/Budibase/budibase/files/7460258/Table.Like.Form-export-1635852719881.txt)

**Reply by ronaldbalza** [reply]
@mjashanks awesome. Thanks for the guidance :) Cheers!

**Reply by melohagan** [reply]
THere's an issue open here for the 'Add new row' part: https://github.com/Budibase/budibase/issues/5658
